fix: Reads UI Windows in the main thread while fetching view hierarchy (#2629)
getsentry/sentry-cocoa: CHANGELOG.md

- Gather profiling timeseries metrics for CPU usage and memory footprint, and thermal and memory pressure events (#2493)

### Fixes

> - Reads UI Windows in the main thread while fetching view hierarchy (#2629)

Reviewer: `m`: Maybe a bit clearer
```suggestion
- Always fetch view hierarchy on the main thread (#2629)
```
